cmd/acr/purge.go

```diff
@@ -264,10 +279,22 @@ func getMatchingRepos(ctx context.Context, repoNames []string, repoRegex string)
 
 // getRepositoryAndTagRegex splits the strings that are in the form <repository>:<regex filter>
 func getRepositoryAndTagRegex(filter string) (string, string, error) {
-	repoAndRegex := strings.Split(filter, ":")
+	// This only selects colons that are not apart of a non-capture group
+	// Note: regexp2 doesn't have .Split support yet, so we just replace the colon with another delimitter \r\n
+	// We choose \r\n since it is an escape sequence that cannot be a part of repo name or a tag
+	noncaptureGroupSupport := regexp2.MustCompile("(?<!\\(\\?):", defaultRegexpOptions)
```

Comment:
With the compatibility mode `regexp2.RE2`, [named ascii character classes](https://github.com/dlclark/regexp2#compare-regexp-and-regexp2) like `[[:alpha:]]`, `[[:digit:]]` are allowed. Are we able to handle those scenarios?

Reply:
It is out of the scope of #96 but it is good scenario to consider.

Reply:
Fixed